### PR TITLE
refactor!: normalize serializers classes, project and packages name

### LINKED ISSUE #136: Standardize serializers classes and packages name
Today we have serializer packages and classes following different naming strategy.

We should standardize the names to follow the same naming strategy.

Proposal:
- ProtobufNetSerializer
- NewtonsoftJsonSerializer
- CoreJsonSerializer (System.Text.Json)
- ConfluentAvroSerializer

**All serializers should be below the KafkaFlow.Serializer namespace directly**

### PR DESCRIPTION
**Pull request by filipeesch**
# Description

Normalize serializers classes, project and packages name to use a more friendly pattern

Fixes #136 

## How Has This Been Tested?

Tested using the unit and integration tests

## Checklist

-   [x] My code follows the style guidelines of this project
-   [x] I have performed a self-review of my own code
-   [ ] I have added tests to cover my changes
-   [ ] I have made corresponding changes to the documentation

### Disclaimer

By sending us your contributions, you are agreeing that your contribution is made subject to the terms of our [Contributor Ownership Statement](https://github.com/Farfetch/.github/blob/master/COS.md)

### CODE REVIEW

File: src/KafkaFlow.Serializer.JsonCore/JsonCoreSerializer.cs [hunk 1]
```diff
@@ -1,28 +1,28 @@
...
+﻿namespace KafkaFlow.Serializer
 {
     using System;
     using System.Text.Json;
 
     /// <summary>
     /// A message serializer using System.Text.Json library
     /// </summary>
-    public class JsonMessageSerializer : IMessageSerializer
+    public class JsonCoreSerializer : IMessageSerializer
```
Comment (dougolima) [inline src/KafkaFlow.Serializer.JsonCore/JsonCoreSerializer.cs:9]: Before we had `XYXMessageSerializer` extending `IMessageSerializer`.

Now we have `XYXSerializer`, so maybe we should rename `IMessageSerializer` to `ISerializer`.

What do you think?

File: src/KafkaFlow.Serializer.SchemaRegistry.ConfluentJson/KafkaFlow.Serializer.SchemaRegistry.ConfluentJson.csproj [hunk 2]
```diff
@@ -1,22 +1,20 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
-  <PropertyGroup>
-    <TargetFramework>netstandard2.0</TargetFramework>
-    <PackageId>KafkaFlow.Serializer.ConfluentJson</PackageId>
-    <Description>Json implementation for KafkaFlow serializer middleware using Confluent.SchemaRegistry.Serdes.Json package</Description>
-    <PackageIconUrl>https://raw.githubusercontent.com/Farfetch/.github/master/images/fuse-logo-128.png</PackageIconUrl>
```
Comment (dougolima) [inline src/KafkaFlow.Serializer.SchemaRegistry.ConfluentJson/KafkaFlow.Serializer.SchemaRegistry.ConfluentJson.csproj:7]: Why was PackageIconUrl removed?

**Review by dougolima — CHANGES_REQUESTED** [review]

File: src/KafkaFlow.Serializer.SchemaRegistry.ConfluentJson/KafkaFlow.Serializer.SchemaRegistry.ConfluentJson.csproj (hunk 2 above)
Comment (filipeesch) [reply src/KafkaFlow.Serializer.SchemaRegistry.ConfluentJson/KafkaFlow.Serializer.SchemaRegistry.ConfluentJson.csproj:7]: I put it in the props file. Doing this way is easier to change in the future e to not forget to add it in new projects

File: src/KafkaFlow.Serializer.JsonCore/JsonCoreSerializer.cs (hunk 1 above)
Comment (filipeesch) [reply src/KafkaFlow.Serializer.JsonCore/JsonCoreSerializer.cs:9]: In the past, I called IMessageSerializer to not create a conflict with the ISerializer of the .NET framework and other things that has this interface defined. I do prefer the ISerializer only too. What do you think?

Comment (dougolima) [reply src/KafkaFlow.Serializer.JsonCore/JsonCoreSerializer.cs:9]: I agree. Let's change to ISerializer so.

File: src/KafkaFlow.Serializer.SchemaRegistry.ConfluentJson/KafkaFlow.Serializer.SchemaRegistry.ConfluentJson.csproj (hunk 2 above)
Comment (dougolima) [reply src/KafkaFlow.Serializer.SchemaRegistry.ConfluentJson/KafkaFlow.Serializer.SchemaRegistry.ConfluentJson.csproj:7]: nice!

File: src/KafkaFlow.Serializer.JsonCore/JsonCoreSerializer.cs (hunk 1 above)
Comment (dougolima) [reply src/KafkaFlow.Serializer.JsonCore/JsonCoreSerializer.cs:9]: It is better to do it when we solve [this ](https://github.com/Farfetch/kafka-flow/issues/142) issue.

**Review by dougolima — APPROVED** [review]

**Pull request merged**